framework/src/modules/interoperability/mainchain/commands/recover_message.ts

```diff
@@ -91,9 +91,11 @@ export class RecoverMessageCommand extends BaseInteroperabilityCommand<Mainchain
 			}
 		}
 
-		// Check that the CCMs are still pending. We can check only the first one,
-		// as the idxs are sorted in ascending order.
-		if (idxs[0] < terminatedOutboxAccount.partnerChainInboxSize) {
+		// as the idxs are sorted in ascending order. Note that one must unset the most significant
+		// bit of an encoded index in idxs in order to get the position in the tree.
+		// See https://github.com/LiskHQ/lips/blob/main/proposals/lip-0031.md#proof-serialization.
+		const firstPosition = parseInt(idxs[0].toString(2).slice(1), 2);
+		if (firstPosition < terminatedOutboxAccount.partnerChainInboxSize) {
 			return {
 				status: VerifyStatus.FAIL,
 				error: new Error('Cross-chain message is not pending.'),
```

Comment:
I think you miss the second check from the LIP: https://github.com/LiskHQ/lips/blob/c24059afc77eebf8288faf49438618c75ea82da6/proposals/lip-0054.md?plain=1#L400

Reply:
The second check will be handled by this PR 👍 
https://github.com/LiskHQ/lisk-sdk/pull/8837